### PR TITLE
Implement ?delays request

### PR DESCRIPTION
**Pull request by bmerry**
There is no proxying via the master controller - that's TBD depending on
what the ICD ends up looking like.

Closes NGC-411.

### CODE REVIEW

File: katsdpcontroller/product_controller.py [hunk 1]
```diff
@@ -1329,6 +1380,26 @@ def must_wait(node):
...
+            stream: product_config.GpucbfAntennaChannelisedVoltageStream,
+            timestamp: aiokatcp.Timestamp,
+            coefficient_sets: Sequence[str]) -> None:
+        n_inputs = len(stream.src_streams)
+        conns = []
+        for i in range(n_inputs // 2):
+            node = self._nodes[f"f.{stream.name}.{i}"]
+            if node.katcp_connection is None:
+                raise FailReply(f"No katcp connection to {node.name}")
+            conns.append(node.katcp_connection)
```
Comment (james-smith-za) [inline katsdpcontroller/product_controller.py:1394]: So the one reservation that I'd have with this, is that if a single f-host wasn't connected, then the entire `?delays` request fails. Is that behaviour that we want?

**Review by james-smith-za — COMMENTED** [review]
Code looks ok to me, just one query regarding the actual behaviour that we want to implement.

I don't see any specification written one way or the other in the ICD, and this implementation may be ok for now, but it's probably something that we are going to want to come back to later.

File: katsdpcontroller/product_controller.py (hunk 1 above)
Comment (bmerry) [reply katsdpcontroller/product_controller.py:1394]: I think it's reasonable until we have health sensors. If we're unable to communicate a delay model to an engine, it should be marked as unhealthy (and maybe eventually killed and restarted).

**Comment by bmerry** [comment]
@james-smith-za gave verbal approval, so merging.

**Pull request merged**